rotation.go

```diff
@@ -309,57 +315,66 @@ func (b *backend) setStaticAccountPassword(ctx context.Context, s logical.Storag
 		defer cancel()
 	}
 
-	// Re-use WAL ID if present, otherwise PUT a new WAL
-	output := &setStaticAccountOutput{WALID: input.WALID}
-
 	config, err := readConfig(ctx, s)
 	if err != nil {
-		return nil, err
+		return output, err
 	}
 	if config == nil {
-		return nil, errors.New("the config is currently unset")
-	}
-
-	newPassword, err := b.GeneratePassword(ctx, config)
-	if err != nil {
-		return nil, err
+		return output, errors.New("the config is currently unset")
 	}
 
-	oldPassword := input.Role.StaticAccount.Password
-
 	// Take out the backend lock since we are swapping out the connection
 	b.Lock()
 	defer b.Unlock()
 
+	var newPassword string
+	if output.WALID != "" {
+		wal, err := b.findStaticWAL(ctx, s, output.WALID)
+		if err != nil {
+			return output, errwrap.Wrapf("error retrieving WAL entry: {{err}}", err)
+		}
+
+		switch {
+		case wal != nil && wal.NewPassword != "":
+			newPassword = wal.NewPassword
```

Comment:
In the case where we want to be updating due to an existing WAL, should we be using `wal.OldPassword`?

Reply:
I'm not following this logic. Wouldn't that simply be a no-op, since the current password would be the OldPassword (because the operation failed, hence the WAL being present).

Reply:
The fact we have a WAL means there was either a scheduled or manually triggered rotation. So it's safest to always try to roll forwards based on the intentions we previously recorded in the WAL. If we rolled back to the old password, we would then want to immediately trigger another attempt at rotating.